src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php

```diff
@@ -83,7 +83,7 @@ public function mapFormsToData($forms, &$data)
             if (null !== $propertyPath && $config->getMapped() && $form->isSubmitted() && $form->isSynchronized() && !$form->isDisabled()) {
                 // If the field is of type DateTime and the data is the same skip the update to
                 // keep the original object hash
-                if ($form->getData() instanceof \DateTime && $form->getData() == $this->propertyAccessor->getValue($data, $propertyPath)) {
+                if ($form->getData() instanceof \DateTime && $form->getData() === $this->propertyAccessor->getValue($data, $propertyPath)) {
```

Comment:
@keradus This is an example, where strict comparison breaks the code: https://3v4l.org/TqWYL

Reply:
nah, this is example where comparision shall be done other way (eg, comparing timestamps)